### PR TITLE
Make `modifySourceData` pass physical column index / property name.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
Currently, the `modifySourceData` hook uses a nonsensical column index (https://jsfiddle.net/js_ziggle/mzw5d7c9/), which can obviously lead to numerous unexpected results.

This PR makes it pass the physical index or a property name as the `column` argument.

### How has this been tested?
Added a test case for #8114.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #8114 

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Review by nikersify — CHANGES_REQUESTED** [review]
I don't know, when the columns are moved the indexes in formulas aren't lining up.

```js
const hot = new Handsontable(example, {
  data: [[1, 2], ['=A1']],
  formulas: { engine: HyperFormula },
  manualColumnMove: [1, 0],
  colHeaders: true,
  contextMenu: true,
  rowHeaders: true,
});

hot.getSourceDataAtCell(1, 1) // "=A1"
hot.getDataAtCell(0, 0) // 2
hot.getDataAtCell(1, 1) // 1 (expected... 2?)
```

I'm not sure if this is what we want to happen in the end, but to be fair the old plugin's column move was way more messed up than this one's now.

File: .changelogs/8114.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "title": "Removed an unwanted column index translation from the `modifySourceData` hook.",
+  "type": "fixed",
+  "issue": 8114,
+  "breaking": false,
```
[the remark this answers is not in the document]
Comment (jansiegel) [reply .changelogs/8114.json:5]: Hm... That's a good point. I _think_ `hot.getDataAtCell(1, 1) // 1` is correct, as it's referencing `=A1`, which is placed at a _physical_ `(1, 1)`, _visual_ `(1, 0)` and we're trying to make HF work exclusively on _physical_ indexes, so moving rows/columns is not recognized at all on HF's side. 
It is really misleading, though. If I remember correctly, that was the reason why we initially decided to stick to _visual_ indexes on this plugin (which came with another set of problems).

That all being said, I'm not exactly sure how it's supposed to work either. Could you help, @wojciechczerniak?

[the remark this answers is not in the document]
Comment (wojciechczerniak) [reply .changelogs/8114.json:5]: You have to sync the move just like all other hooks

```ts
this.addHook('beforeColumnMove', (...args) => {
  if (!this.engine.isItPossibleToMoveColumns( ...args )) {
    return false;
  }
});

this.addHook('afterColumnMove', (...args) => {
  this.engine.moveColumns( ...args )
});
```

File: src/plugins/formulas/formulas.js [hunk 2]
```diff
@@ -28,7 +32,9 @@ Hooks.getSingleton().register('afterSheetRenamed');
 Hooks.getSingleton().register('afterFormulasValuesUpdate');
 
 /**
- * This plugin allows you to perform Excel-like calculations in your business applications. It does it by an integration with our other product, [HyperFormula](https://handsontable.github.io/hyperformula/), which is a powerful calculation engine with an extensive number of features.
+ * This plugin allows you to perform Excel-like calculations in your business applications. It does it by an
+ * integration with our other product, [HyperFormula](https://handsontable.github.io/hyperformula/), which is a
```
Comment (krzysztofspilka) [inline src/plugins/formulas/formulas.js]: @jansiegel better URL: handsontable.com/docs/hyperformula

File: src/plugins/formulas/formulas.js [hunk 3]
```diff
@@ -341,6 +350,41 @@ export class Formulas extends BasePlugin {
     });
   }
 
+  /**
+   * Recreates the initial settings of the plugins bupassed by the engine logic.
```
Comment (krzysztofspilka) [inline src/plugins/formulas/formulas.js:494]: bupassed > bypassed

**Comment by jansiegel** [comment]
Closing as it's been on hold for almost two years and has become outdated.

**Pull request closed without merging**